### PR TITLE
Add tests for button component

### PR DESCRIPTION
**Pull request by L1fescape**
Let me know if there are any additional tests you'd like implemented! I was unable to test default and disabled style values because `StyleSheet.create` returns a hash of references to the classnames generated for those styles rather than the actual style values. Idk a way around that.

### CODE REVIEW

File: src/components/Button/__tests__/index-test.js [hunk 1]
```diff
@@ -1,5 +1,52 @@
...
+import React from 'react';
+import Button from '..';
+import { mount, shallow } from 'enzyme';
+
+const findNativeButton = wrapper => wrapper.find('[role="button"]');
+
 describe('components/Button', () => {
-  test.skip('NO TEST COVERAGE', () => {});
+  describe('rendered element', () => {
+    test('is a "div"', () => {
```
Comment (necolas) [inline src/components/Button/__tests__/index-test.js]: This test is not necessary, as it is testing the implementation details of `Button` dependencies

File: src/components/Button/__tests__/index-test.js [hunk 2]
```diff
@@ -1,5 +1,52 @@
 /* eslint-env jasmine, jest */
 
+import React from 'react';
+import Button from '..';
+import { mount, shallow } from 'enzyme';
+
+const findNativeButton = wrapper => wrapper.find('[role="button"]');
```
Comment (necolas) [inline src/components/Button/__tests__/index-test.js]: Can you use something like `getDOMNode` instead of leaking knowledge about how `accessibilityRole` is implemented into this test?

File: src/components/Button/__tests__/index-test.js [hunk 3]
```diff
@@ -1,5 +1,52 @@
...
+  });
+
+  test('prop "color"', () => {
+    const onPress = () => {}
+    const color = 'blue';
+    const button = findNativeButton(mount(<Button color={color} onPress={onPress} title="" />));
+    expect(button.prop('style')).toEqual({backgroundColor: color});
+  });
+
+  test('prop "disabled"', () => {
```
Comment (necolas) [inline src/components/Button/__tests__/index-test.js]: Is this really testing anything?

File: src/components/Button/index.js [hunk 4]
```diff
@@ -28,7 +28,8 @@ class Button extends Component {
   };
 
   render() {
-    const { accessibilityLabel, color, disabled, onPress, testID, title } = this.props;
+    const { accessibilityLabel, color, onPress, testID, title } = this.props;
+    const disabled = this.props.disabled !== undefined ? this.props.disabled : 'disabled' in this.props;
```
Comment (necolas) [inline src/components/Button/index.js]: Why this change?

Comment (L1fescape) [inline src/components/Button/index.js]: To match the html spec for boolean attributes. The presence of a boolean attribute results in a true value regardless of what that value is set to, which I now realize means `diabled={false}` should also equal true. This is probably undesirable behavior so I'll just revert this change.

File: src/components/Button/__tests__/index-test.js (hunk 3 above)
Comment (L1fescape) [inline src/components/Button/__tests__/index-test.js]: Haha nope, not really. I could update it to test that the `disabled` prop prevents the button from being pressed:

```javascript
  test('prop "disabled" should prevent "onPress" from firing', () => {
    const onPress = jest.fn();
    const component = shallow(<Button disabled={true} onPress={onPress} title="" />);
    component.simulate('press');
    expect(onPress).not.toHaveBeenCalled();
  });
```

however I'm getting this failure when I do that: 

```
expect(jest.fn()).not.toHaveBeenCalled()

    Expected mock function not to be called but it was called with:
      Array []
```

Could this be an issue with simulating press events in enzyme?

File: src/components/Button/__tests__/index-test.js [hunk 5]
```diff
@@ -1,5 +1,30 @@
...
+    const text = 'Click me';
+    const component = mount(<Button onPress={onPress} title={text} />);
+    expect(component.text()).toEqual(text);
+  });
+
+  test('prop "color"', () => {
+    const onPress = () => {}
+    const color = 'blue';
+    const button = findNativeButton(mount(<Button color={color} onPress={onPress} title="" />));
+    expect(button.style.backgroundColor).toEqual(color);
```
Comment (L1fescape) [inline src/components/Button/__tests__/index-test.js]: Would you prefer this test use snapshots instead?

File: src/components/Button/__tests__/index-test.js [hunk 6]
```diff
@@ -1,5 +1,30 @@
...
+  });
+
+  test('prop "onPress"', () => {
+    const onPress = jest.fn();
+    const component = shallow(<Button onPress={onPress} title="" />);
+    component.simulate('press');
```
Comment (doxiaodong) [inline src/components/Button/__tests__/index-test.js]: @L1fescape Sorry for the question.  How do do make press simulate work in enzyme or jest?
